### ISSUE TITLE
AureliaHUB leaks memory

### ISSUE THREAD

**Issue by peteigel** [aurelia/framework#800]
**I'm submitting a bug report**

* **Library Version:**
N/A

**Please tell us about your environment:**
* **Operating System:**
OSX 10.12.5

* **Node Version:**
N/A

* **NPM Version:**
N/A

* **Aurelia CLI OR JSPM OR Webpack AND Version**
N/A

* **Browser:**
Chrome 59, Firefox 54

* **Language:**
N/A

**Current behavior:**
When navigating around AureliaHUB, memory usage appears to grow without clear bounds.

I wrote [a script](https://gist.github.com/peteigel/5b67ff0bc1e47228fbc3ada6c3b5d975) for the browser console to simulate navigating around in circles. In both Chrome and Firefox, I observed memory usage growing slowly.

Below is a screenshot of Chrome's Allocation Timeline profile:
![Chrome Allocation Timeline](https://user-images.githubusercontent.com/4098889/28139933-d2b89f46-671b-11e7-8c62-35e521f5d5e3.png)

And Before/After Heap Snapshots in Firefox:
**Before**: 8.39MB
![FF Heap Before](https://user-images.githubusercontent.com/4098889/28140015-36b82cbe-671c-11e7-8cc1-d14820a4a1ac.png)

**After**: 57.20MB
![FF Heap After](https://user-images.githubusercontent.com/4098889/28140109-8eaa1a5e-671c-11e7-8913-ab8545240ac2.png)

The after snapshot was taken after ~500 simulated mouse clicks. The navigations item were the first and only set I tried, I suspect this issue is site-wide.

**Expected/desired behavior:**
Memory usage should not continue to grow when navigating to pages repeatedly.

* **What is the motivation / use case for changing the behavior?**
I took these measurements after my team and I spent a long time looking into a memory leak in our application. We were not able to track down a cause within our code, so I tested AureliaHUB as an example of another non-trivial Aurelia App.

----------

The leaked objects appear to have the `e` prototype in the minified code. I believe they may be `Binding` objects, but I am not sure.

**Comment by AshleyGrant** [comment]
@EisenbergEffect this looks like it's a possible bug in the framework, not the HUB.

**Comment by EisenbergEffect** [comment]
To be honest, the HUB is a terrible piece of code. We're replacing it and the entire web site soon, hopefully in another month or so. The whole thing has been written from scratch: all new html, js and css. I'm closing for now. If we find a general memory leak, let's open that somewhere with a simple repro.

**Comment by harkenn** [comment]
....so are we expected to go out and find exactly where the resource leak is in the framework? I don't see any followup issue created to investigate this and @EisenbergEffect 's quick closure of this ticket mere hours after its opening leaves me uneasy that this isn't a priority for Aurelia. The scale we see this leak occurring at, its quickly becoming a huge issue for the b2b applications we're developing that (for the moment) use Aurelia.

**Comment by AshleyGrant** [comment]
@harkenn I believe that Rob is asking to have a separate issue filed for this with a simple repro of the leak. Probably something that uses one of the skeleton projects would work best.
